components/proxy/src/main/kotlin/com/hotels/styx/services/OriginsAdminHandler.kt

```diff
@@ -0,0 +1,168 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.common.net.MediaType.JSON_UTF_8
+import com.hotels.styx.*
+import com.hotels.styx.api.*
+import com.hotels.styx.api.HttpMethod.*
+import com.hotels.styx.api.HttpResponseStatus.*
+import com.hotels.styx.infrastructure.configuration.json.mixins.ErrorResponseMixin
+import com.hotels.styx.routing.RoutingObjectRecord
+import com.hotels.styx.routing.db.StyxObjectStore
+import java.lang.RuntimeException
+import java.nio.charset.StandardCharsets.UTF_8
+
+/**
+ * Handles URLs like: /admin/providers/<provider-name>/<app-id>/<origin-id>/state
+ * The parent will pass anything like /admin/providers/provider-name/xxx... to this handler
+ * This handler is configured with the base path /admin/providers/provider-name so that it can extract the remainder
+ * One handler for all origins
+ */
+internal class OriginsAdminHandler(
```

Comment:
Please use `UrlPatternRouter` to implement this handler. It provides most of the required functionality out of the the box:

* Match on HTTP methods
* Match on URL paths
* Extract variables from URL paths

Reply:
I completely missed that class. I'll re-implement. Thanks.